common/cache/lru_test.go

```diff
@@ -384,3 +384,74 @@ func TestPanicOptionsIsNil(t *testing.T) {
 
 	New(nil)
 }
+
+func TestEvictItemsPastTimeToLive_AllExpired(t *testing.T) {
```

Comment:
we are going to slowly switch to table driven tests. please rewrite these new tests using that pattern https://golang.cafe/blog/golang-table-test-example

Reply:
I would say table tests work very well compared to the other approaches. Some benefits are
- Maintainability: For the same unit of work you write the body once and you can easily add more cases without making redundant copy of the validation logic. If you need to change something in your test body you have to change it in N different places without table driven tests.
- Readability: Creating structs for test cases makes it very clear to understand what are the inputs and expected outputs. Much better readability.

Scope of your test function and what you put in the test body can of course increase complexity but that can and does happen in other approaches. [Style guide](https://github.com/uber-go/guide/blob/master/style.md#avoid-unnecessary-complexity-in-table-tests) has a good point about not combining irrelevant cases together.

Reply:
From my experience:
Table test works great for a set of similar cases that can be easily defined.
But every complication requires modification of initial struct and very fast it becomes hard to manage.
Lru is pretty complicated and codiying steps will be harder than just expressing what you are trying. At least the way I see how it could be codified.

Reply:
To be clear there's no suggestion to fit all of the lru_test.go into single tabular function to simulate all the scenarios.
Different scenarios deserve their own functions which simulate something we are interested to validate.

Regarding the new functions in this PR, all lines are duplicate except ~1-2 line difference (either an extra sleep or calling EvictItemsPastTimeToLive vs Size). This can be turned into table format easily.